src/App.context.js

```diff
@@ -50,8 +52,21 @@ export function AppContextProvider({
     organization,
   ]);
 
+  useEffect( () => {
+    async function getLanguages() {
+      const langs = (await fetch('https://td.unfoldingword.org/exports/langnames.json'))
```

Comment:
also, if this JSON was in an RCL, it would make it easy to create a "BUILD SCRIPT" that would DOWLNOAD the JSON and construct a real type...
This could be done at BUILD time (for the RCL) to avoid (1) "live" downloading the JSON file and (2) deserializing the JSON.

Just an idea.  I realize this does not accomplish the job of reading "LIVE" data.  But in order to update the data, you would simply release another version of the RCL.

If semver filtering is enabled ^1.0.0, then the new langnames file is available to the app  next time the app is built.

But I'm not even sure how much overhead is involved in downloading/deserializing the data.

BUT it perhaps works against the pending goals of "offline" or "occasionally offline" tools.

Reply:
Just ideas!   Maybe we can chat

Reply:
The only thing that would be appropriate for inclusion in a separate package would be the `getLanguages()` function, right? That is only 5 lines of code... is it really worth the effort?

Also consider that Klappy wanted the 'gitea-react-toolkit' to be free of uW specific code. And this is quite uW specific. So if not GRT, where would this code fit?

I agree with your point on offline support. But maybe tc-create-app will be superseded by GWE by that time. Plus Next.js has a way of fetching data at build time which would be even better.

At any rate, we can meetup and discuss more.